wallet/src/shared/cryptography/mnemonics.ts

```diff
@@ -1,10 +1,12 @@
 // Copyright (c) 2022, Mysten Labs, Inc.
 // SPDX-License-Identifier: Apache-2.0
 
+import { HDKey } from '@scure/bip32';
 import bip39 from 'bip39-light';
+import { derivePath, getPublicKey } from 'ed25519-hd-key';
 import nacl from 'tweetnacl';
 
-import type { Ed25519KeypairData } from '@mysten/sui.js';
+import type { Ed25519KeypairData, Secp256k1KeypairData } from '@mysten/sui.js';
```

Comment:
the added the functions here is quite verbose to differentiate ed25519 vs secp256k1. this should really belong to the corresponding keypair file. will follow up a PR to move this class into sdk/typescript/cryptography